### PR TITLE
Feature/textarea counter

### PR DESCRIPTION
**Pull request by Laurin-migi**

### CODE REVIEW

File: src/es/components/molecules/form/Form.js [hunk 1]
```diff
@@ -269,6 +331,15 @@ export default class Form extends Shadow() {
     }))
   }
 
+  updateCounter (textArea) {
+    if (!textArea) return
```
Comment (Weedshaker) [inline src/es/components/molecules/form/Form.js:335]: !textArea || testArea.value === undefined

File: src/es/components/molecules/form/Form.js [hunk 2]
```diff
@@ -65,17 +67,32 @@ export default class Form extends Shadow() {
...
+      this.updateCounter(this.root.querySelector(':focus'))
+    }
   }
 
   connectedCallback () {
     if (this.shouldComponentRenderCSS()) this.renderCSS()
+    if (this.shouldComponentRenderHTML()) this.renderHTML()
     if (this.submit) this.submit.addEventListener('click', this.clickListener)
     if (this.hasAttribute('use-recaptcha') !== null) this.addEventListener('submit', this.submitListener)
+    this.textarea.forEach(a => {
```
Comment (Weedshaker) [inline src/es/components/molecules/form/Form.js:81]: you added the event listener multiple times, for each textarea. Avoid this and only add it once on document, expl: https://github.com/mits-gossau/web-components-toolbox/blob/master/src/es/components/atoms/input/Input.js#L65

document.addEventListener('keyup', this.keydownListener)

just search if there is one element with maxlength and without no-counter, to figure out if you got to add the listener. Something like:

this.root.querySelector('[maxlength]:not([no-counter])')

File: src/es/components/molecules/form/Form.js [hunk 3]
```diff
@@ -65,17 +67,32 @@ export default class Form extends Shadow() {
...
+      if (a.hasAttribute('maxlength') && !a.hasAttribute('no-counter')) {
+        a.addEventListener('keyup', this.textAreaKeyUpListener)
+      }
+    })
   }
 
   disconnectedCallback () {
     if (this.submit) this.submit.removeEventListener('click', this.clickListener)
     if (this.hasAttribute('use-recaptcha') !== null) this.removeEventListener('submit', this.submitListener)
+    this.textarea.forEach(a => {
```
Comment (Weedshaker) [inline src/es/components/molecules/form/Form.js:91]: same here, only add the listener once

File: src/es/components/molecules/form/Form.js [hunk 4]
```diff
@@ -87,6 +104,15 @@ export default class Form extends Shadow() {
     return !this.root.querySelector(`:host > style[_css], ${this.tagName} > style[_css]`)
   }
 
+  /**
+   * evaluates if a render is necessary
+   *
+   * @return {boolean}
+   */
+  shouldComponentRenderHTML () {
+    return !this.render
```
Comment (Weedshaker) [inline src/es/components/molecules/form/Form.js:113]: isn't there a better way than just a flag to figure out if the html render ran?

File: src/es/components/molecules/form/Form.js [hunk 5]
```diff
@@ -178,6 +205,17 @@ export default class Form extends Shadow() {
         padding: 0 var(--content-spacing);
         max-width: 30vw !important;
       }
+      .umbraco-forms-field-wrapper{
```
Comment (Weedshaker) [inline src/es/components/molecules/form/Form.js:208]: does this no change every field even without maxlength? no side effects?

**Review by Weedshaker — COMMENTED** [review]
nice work, please correct the few issues.

**Review by Weedshaker — CHANGES_REQUESTED** [review]
correct, thx

File: src/es/components/molecules/form/Form.js (hunk 4 above)
Comment (Laurin-migi) [reply src/es/components/molecules/form/Form.js:113]: Nicht dass ich wüsste. Es wird ja nicht unbedingt etwas im HTML hinzugefügt.

File: src/es/components/molecules/form/Form.js (hunk 5 above)
Comment (Laurin-migi) [reply src/es/components/molecules/form/Form.js:208]: No side effects, everything works the same

File: src/es/components/molecules/form/Form.js (hunk 2 above)
Comment (Laurin-migi) [reply src/es/components/molecules/form/Form.js:81]: Mostly only one textarea

File: src/es/components/molecules/form/Form.js (hunk 4 above)
Comment (Laurin-migi) [reply src/es/components/molecules/form/Form.js:113]: with querrySelector

**Review by Weedshaker — APPROVED** [review]

**Review by Weedshaker — COMMENTED** [review]
great

**Pull request merged**